src/workerd/io/actor-cache.c++

```diff
@@ -2139,6 +2142,37 @@ void ActorCache::ensureFlushScheduled(const WriteOptions& options) {
   }
 }
 
+kj::Maybe<kj::Promise<void>> ActorCache::onNoPendingFlush() {
+  // This function returns a Maybe<Promise> because a falsy maybe allows the jsg interface to make
```

Comment:
Just to make sure I understand, this goal of returning a resolved `jsg::Promise` is the reason for tracking `flushesEnqueued` rather than just always returning `lastFlush.addBranch()`?

That's great to be clear, I just want to make sure there isn't something more I'm missing.